src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -83,7 +83,8 @@ private String requestToString(HttpServletRequest request) throws IOException {
   /**
    * Calculates sentiment score of text. The score is from -1 (negative) to +1 (positive).
    */ 
-  private Float calculateSentimentScore(String text) {
+  private Float calculateSentimentScore(String text) throws ApiException, IOException {
+    LanguageServiceClient languageService = createLanguageServiceClient();
```

Comment:
just want to make sure I understand the problem. Are we not able to reuse instances of LanguageServiceClient?

Reply:
Yeah if you try and reuse it you'll run into a RejectedExecutionException

Reply:
After reading the documentation I think the problem is not that we can't call languageService.analyzeSentiment more than once per instance. I think the problem is that we shouldn't call languageService.close until we're done with the instance.

The original implementation calls LanguageService.create in the constructor, then calls languageService.close (line 94 in this revision) in the first call to calculateSentimentScore. The documentation [1] says that future behavior is undefined. So in the next call to calculateSentimentScore we can't expect reasonable behavior from languageService.

Instead of creating a new client for each request, can we simply delete the call to close?

[1] https://googleapis.dev/nodejs/language/latest/v1.LanguageServiceClient.html#close

Reply:
@ajpaon I was concerned about leaving an open connection indefinitely (while the server is running). I feel like LanguageService API might force close the connection after some timeout.

Reply:
It's fine to keep a grpc channel open indefinitely. Here's a thread on the grpc-java repo that discusses this: https://github.com/grpc/grpc-java/issues/3268